### PR TITLE
Rae2019compressive 2

### PR DESCRIPTION
**Pull request by specfazhou**
this is my first draft of Rae2019compressive_2 summary

### CODE REVIEW

File: _summaries/Rae2019compressive_2.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+layout: summary
+title: Summary
+paper: Rae2019compressive_2
+# Please fill out info below
+author: specfazhou
+score: 8/10
+---
+
+## core idea
+In this article, the authors proposed Compressive Transformer, a variant of Transformer, which achieved the-state-of-art resutls on character-based language modelligng and word-level language modelling.
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:11]: Typo: "results", "modelling"

File: _summaries/Rae2019compressive_2.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Compressive Transformer also performed very well on the book-level language modelling. The Compressive Transformer, neither like Transformer that computationally expensive, nor like TransformerXL that discard old enough activations, it compress past activations into compressed memory.
+The Compressive Transformer not only achieved computational and storage efficiency, but also remember all the past activations. 
+
+## how is it realized?
+The Compressive Transformer only uses attention to propagate information, and it has two parts of memories --- respective memory (FIFO memory) and compressed memory. Sequences first come to the respective memory, and with newer sequences come in, old activations are compressed to compressed memory by using some compression functions. In other words, memories obtained recently are make up larger storage while all the old memories make up small part of storage. The following algorithm describes this process:
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:16]: Maybe you can mention that both respective memory and compressed memory are FIFO, but they differ in the way old memories are handled.  For respective memory, the $$c$$ oldest entries are compressed into a single entroy in the compressed memory.  And compressed memory just discards the old entries.

File: _summaries/Rae2019compressive_2.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Compressive Transformer also performed very well on the book-level language modelling. The Compressive Transformer, neither like Transformer that computationally expensive, nor like TransformerXL that discard old enough activations, it compress past activations into compressed memory.
+The Compressive Transformer not only achieved computational and storage efficiency, but also remember all the past activations. 
+
+## how is it realized?
+The Compressive Transformer only uses attention to propagate information, and it has two parts of memories --- respective memory (FIFO memory) and compressed memory. Sequences first come to the respective memory, and with newer sequences come in, old activations are compressed to compressed memory by using some compression functions. In other words, memories obtained recently are make up larger storage while all the old memories make up small part of storage. The following algorithm describes this process: 
+<img width = "600" alt = "compressive_transformer" src = "compressive_transformer.png">
+<img width = "600" alt = "compressive_transformer_algo1" src = "compressive_transformer_algo1.png">
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:18]: I'm not sure if the algorithm block really helps the reader.  There are too many variables that need to be defined.  Also is line 11 wrong?  If I understood the paper correctly, shouldn't the left-hand side be $$cm_{t+1}^{(i)}$$?  If you think this is wrong, you could mention that as a mistake in the paper.

File: _summaries/Rae2019compressive_2.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## technical details
+There are four choices for compression functions, which are max/mean pooling, 1D convolution, dilated convolutions, and "most-used"。 Here, the "most-used" function makes memory most used to be preserved. All compression functions can be trained.  As for the loss, there are two supplementary losses should be considered ---- auto-encoding loss, which recover compressed memories, and attention-reconstruction loss, which recover the context-based attention from respective memories and compressed memories. Attention-reconstruction loss is a bit complicated, as follows:
+
+<img width = "600" alt = "compressive_transformer_algo2" src = "compressive_transformer_algo2.png">
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:23]: It'd be nice if you could distill the intuition of this loss.  It's essentially using reconstruction loss on the output of attention, rather than the memory entries themselves directly.

File: _summaries/Rae2019compressive_2.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+# results
+It seems the Compressive Transformer outperforms TransformerXL on PG-19, which is a large datasets with datas are long-context extracted from project Gutenberg. the Compressive Transformer model also achieved the-state-of-art resutls on Enwik8, and if people use 1D convolution compression function together with attention reconstruction loss function, the result will be the best among all other Compressive Transformer models. The Compressive Transformer also did a good job in speech, comparable to the-state-of-art model, but more parameters efficient. The authors also found that the older memories are not less frequently accessed.
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:26]: You could maybe split this into two sections: one for the main benchmark results (e.g. PG-19, Enwik, Wikitext), and one for ablation studies (e.g. which combination of compression loss function as well as compression layer worked the best).

File: _summaries/Rae2019compressive_2.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+<img width = "600" alt = "compressive_transformer_algo2" src = "compressive_transformer_algo2.png">
+
+# results
+It seems the Compressive Transformer outperforms TransformerXL on PG-19, which is a large datasets with datas are long-context extracted from project Gutenberg. the Compressive Transformer model also achieved the-state-of-art resutls on Enwik8, and if people use 1D convolution compression function together with attention reconstruction loss function, the result will be the best among all other Compressive Transformer models. The Compressive Transformer also did a good job in speech, comparable to the-state-of-art model, but more parameters efficient. The authors also found that the older memories are not less frequently accessed. 
+
+# three points I like
+1. Compressive Transformer compress old memories rather than discard it
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:29]: Perhaps a more accurate statement is that it maintains compressed versions of memory entries, thus extending a portion of its memory by a predefined factor (the $$c$$ hyperparameter).

File: _summaries/Rae2019compressive_2.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+2. Compressive Transformer doesn't keep all activations, but compress them, which let it becomes more storage efficient
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:30]: Isn't this point basically the same as the first point?  Maybe you can merge them into one.

File: _summaries/Rae2019compressive_2.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+3. attention increased from the oldest activations stored in respective memory to the compressed memory.
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:31]: I'm not sure what this sentence means.  In what sense did attention "increase"?

File: _summaries/Rae2019compressive_2.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+title: Summary
+paper: Rae2019compressive_2
+# Please fill out info below
+author: specfazhou
+score: 8/10
+---
+
+## core idea
+In this article, the authors proposed Compressive Transformer, a variant of Transformer, which achieved the-state-of-art resutls on character-based language modelligng and word-level language modelling.
+Compressive Transformer also performed very well on the book-level language modelling. The Compressive Transformer, neither like Transformer that computationally expensive, nor like TransformerXL that discard old enough activations, it compress past activations into compressed memory.
```
Comment (jaywhang) [inline _summaries/Rae2019compressive_2.md:12]: I don't think you can say that Compressive Transformer remembers "all past activations".

File: _summaries/Rae2019compressive_2.md (hunk 1 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:11]: thanks

File: _summaries/Rae2019compressive_2.md (hunk 2 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:16]: It seems in the paper, it writes, "The key principle of the Compressive Transformer is to compress these old
memories, instead of discarding them, and store them in an additional compressed memory." I think the compressed memories will not be discarded

File: _summaries/Rae2019compressive_2.md (hunk 3 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:18]: Sure, I will take a careful look at the algorithm. If it is wrong, I will mention it.

File: _summaries/Rae2019compressive_2.md (hunk 9 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:12]: Can you tell me what part of the paper mentions compressed memory will be discarded?

File: _summaries/Rae2019compressive_2.md (hunk 8 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:31]: I think there is a graph, I think it is a good idea if I show the picture during my presentation, thank you for pointing out

File: _summaries/Rae2019compressive_2.md (hunk 7 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:30]: Yeah, I agree. I merged them together, added one more point, thanks

File: _summaries/Rae2019compressive_2.md (hunk 4 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md:23]: Nice, I will add this point to my summary.

File: _summaries/Rae2019compressive_2.md (hunk 5 above)
Comment (specfazhou) [reply _summaries/Rae2019compressive_2.md]: that's definitely better, thanks!

**Review by zhaoyue-zephyrus — APPROVED** [review]
